test/e2e/utils.go

```diff
@@ -7,13 +7,35 @@ import (
 
 	operator "github.com/mattermost/mattermost-operator/pkg/apis/mattermost/v1alpha1"
 
+	mysqlOperator "github.com/presslabs/mysql-operator/pkg/apis/mysql/v1alpha1"
 	appsv1 "k8s.io/api/apps/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/wait"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+func waitForMySQLStatusReady(t *testing.T, dynclient client.Client, namespace, name string, replicas int, retryInterval, timeout time.Duration) error {
+	mysql := &mysqlOperator.MysqlCluster{}
+	err := wait.Poll(retryInterval, timeout, func() (done bool, err error) {
+		errClient := dynclient.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: namespace}, mysql)
+		if errClient != nil {
+			return false, errClient
+		}
+
+		if mysql.Status.ReadyNodes == replicas {
+			return true, nil
+		}
+		t.Logf("Waiting for MySQL cluster ReadyNodes: %d\n", mysql.Status.ReadyNodes)
+		return false, nil
+	})
+	if err != nil {
+		return err
+	}
+	t.Logf("Cluster has %d ready nodes!\n", mysql.Status.ReadyNodes)
```

Comment:
nit: this would be a bit clearer I think
```suggestion
	t.Logf("All MySQL cluster nodes (%d) are ready!\n", mysql.Status.ReadyNodes)
```